src/file_process.cpp

```diff
@@ -0,0 +1,117 @@
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <fcntl.h>
+#include <errno.h>
+#include <regex>
+#include <bits/stdc++.h> 
+#include <boost/algorithm/string.hpp>
+
+#include "../include/file_process.hpp"
+
+#define BUF_SIZE 2048
+
+using namespace std;
+
+void output_file_process(char *fin, char *fout) {
+    // check if the read access to the input file is OK
+    if (access(fin, R_OK)) {
+        std::cout << "The file " << fin << " can not be read from, errno = " << errno << "\n";
+    }
+    else {
+        // read the input file into a string
+        char *file = read_file(fin);
+        // write the data from the input file string into output file
+        write_file(fout, file);
+    }
+}
+
+char *read_file(char *fin) {
+    char read_buf[BUF_SIZE];
+    char *file;
+    int size;
+    int i = 1;
+	
+    int fdi = open(fin, O_RDONLY);
+	
+    if (fdi > 0) {
+        // read a block of data
+        while ((size = read(fdi, read_buf, BUF_SIZE)) != 0) {
+            if (i == 1) {
+                // allocate memory and store the first block of data
+                file = (char *) malloc(sizeof(char) * BUF_SIZE);
+                file = strncpy(file, read_buf, size);
+            }
+            else {
+                // reallocate memory and append the next block of data
+                file = (char *) realloc(file, sizeof(char) * BUF_SIZE * i);
+                strncpy(file + BUF_SIZE * (i - 1), read_buf, size);
+            }
+            i++;
+        }
+    }
+    else {
+        cout << "The file " << fin << " can not be opened for reading, errno = " << errno << "\n";
+    }
+	
+    int fc = close(fdi);
+    if (fc < 0) {
+        cout << "The file " << fin << " can not be closed\n";
+    }
+	
+    return(file);
+}
+
+void write_file(char *fout, char *file) {
+    ofstream out_file (fout);
+    char* pch = NULL;
+    char* time;
+	
+    if (out_file.is_open()) {
```

Comment:
Hi,

Indentation not good when viewed in the local text editor

![agarwal](https://user-images.githubusercontent.com/46407088/63226351-ac3e3280-c1a6-11e9-9a06-e49f3e9a166d.jpg)

Reply:
solved